src/main/java/com/jcabi/aether/Repository.java

```diff
@@ -120,13 +137,31 @@ public RemoteRepository remote() {
         remote.setUrl(this.url);
         remote.setPolicy(false, this.release);
         remote.setPolicy(true, this.snapshot);
-        remote.setProxy(this.proxy);
-        remote.setAuthentication(this.authentication);
+        Authentication auth = null;
+        if (this.authentication != null) {
+            auth = new Authentication(
+                this.authentication.getUsername(),
+                this.authentication.getPassword(),
+                this.authentication.getPrivateKeyFile(),
+                this.authentication.getPassphrase()
+            );
+        }
+        remote.setAuthentication(auth);
+        Proxy proxy = null;
+        if (this.repoproxy != null) {
+            proxy = new Proxy(
+                this.repoproxy.getType(),
+                this.repoproxy.getHost(),
+                this.repoproxy.getPort(),
+                auth
+            );
+        }
+        remote.setProxy(proxy);
```

Comment:
similarly here: move it into the `if` and set directly: `remote.setProxy(new Proxy(`